### PR TITLE
Fix #determine_target_shipment call

### PR DESCRIPTION
**Pull request by stem**
solidusio/solidus#3197 add a mandatory parameter which we should send

As I understand the rest of the code, we should send the missing quantity, not
the total quantity but I'm not completely sure about that.

### CODE REVIEW

File: app/models/spree/order_inventory_assembly.rb [hunk 1]
```diff
@@ -21,8 +21,9 @@ def verify(shipment = nil)
           self.variant = part
 
           if existing_parts < total_parts
-            shipment ||= determine_target_shipment
-            add_to_shipment(shipment, total_parts - existing_parts)
+            quantity = total_parts - existing_parts
+            shipment ||= determine_target_shipment(quantity)
```
Comment (filippoliverani) [inline app/models/spree/order_inventory_assembly.rb:25]: I think that to support every Solidus version we should handle both cases with something like:
```ruby

      def determine_target_shipment(quantity)
        if determine_target_shipment_requires_quantity?
          super(quantity)
        else
          super()
        end
      end

      private

      def determine_target_shipment_requires_quantity?
         OrderInventory.instance_method(:determine_target_shipment).arity != 0
      end
    end
```

**Comment by spaghetticode** [comment]
I had a PR for this, but it was closed automatically by the stale bot and I did not catch it in the emails 🤦... see https://github.com/solidusio-contrib/solidus_product_assembly/pull/50. 

@stem I think you can take inspiration from the code in that PR, as it addresses Filippo's concern.

**Comment by stem** [comment]
@spaghetticode thanks for the pointer, I didn't looked at closed but not merged PR before writing this one. As you have already done the work, you should be the author 😉 
WDYT about closing this one and reopening #50 which seems complete and backward compatible?

**Comment by aldesantis** [comment]
I've merged @spaghetticode's PR. 🙂

**Pull request closed without merging**